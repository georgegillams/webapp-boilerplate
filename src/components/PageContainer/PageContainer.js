import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { cssModules } from 'gg-components/helpers/cssModules';

import STYLES from './page-container.scss';
import { DEBUG_SHOW_PAGE_CONTAINER_KEY } from 'helpers/storageConstants';

export const LAYOUT_STYLES = {
  default: 'default',
  prose: 'prose',
  fullWidthCenter: 'fullWidthCenter',
};

const getClassName = cssModules(STYLES);

const PageContainer = props => {
  const { className, layout, ...rest } = props;
  const [showDebug, setShowDebug] = useState(false);

  useEffect(() => {
    setShowDebug(window.localStorage.getItem(DEBUG_SHOW_PAGE_CONTAINER_KEY) === 'true');
  }, []);

  const outerClassNames = [getClassName('page-container__container')];
  if (showDebug) {
    outerClassNames.push([getClassName('page-container__container--debug')]);
  }
  if (layout === LAYOUT_STYLES.default) {
    outerClassNames.push([getClassName('page-container__container--regular')]);
  }
  if (layout === LAYOUT_STYLES.prose) {
    outerClassNames.push([getClassName('page-container__container--prose')]);
  }
  if (layout === LAYOUT_STYLES.fullWidthCenter) {
    outerClassNames.push([getClassName('page-container__container--full-width-center')]);
  }

  if (className) {
    outerClassNames.push(className);
  }

  return <main className={outerClassNames.join(' ')} {...rest} />;
};

PageContainer.propTypes = {
  layout: PropTypes.oneOf(Object.keys(LAYOUT_STYLES)),
  className: PropTypes.string,
};

PageContainer.defaultProps = {
  layout: LAYOUT_STYLES.default,
  className: null,
};

export default PageContainer;
