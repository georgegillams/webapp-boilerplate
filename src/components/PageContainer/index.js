export { default, LAYOUT_STYLES } from './PageContainer';
