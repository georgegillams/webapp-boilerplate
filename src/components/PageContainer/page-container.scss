@import '~utils/tokens';

.page-container {
  &__container {
    display: flow-root;
    width: 100%;
    max-width: 60rem;
    margin-right: auto;
    margin-left: auto;
    padding-right: 2rem;
    padding-left: 2rem;

    @include bpk-breakpoint-tablet {
      padding-right: 1rem;
      padding-left: 1rem;
    }

    &--debug {
      background-color: orange;

      @media (prefers-color-scheme: dark) {
        background-color: darkorange;
      }
    }
  }
}
