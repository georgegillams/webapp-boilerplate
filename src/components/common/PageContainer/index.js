export { default, WIDTHS } from './PageContainer';
