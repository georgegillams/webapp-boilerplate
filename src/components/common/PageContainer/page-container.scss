@import '~utils/tokens';

.page-container {
  &__container {
    display: flow-root;
    width: 100%;
    margin-right: auto;
    margin-left: auto;

    &--debug {
      background-color: orange;

      @media (prefers-color-scheme: dark) {
        background-color: darkorange;
      }
    }

    &--regular {
      max-width: $page-max-width;
      padding-right: $spacing-lg;
      padding-left: $spacing-lg;

      @include bpk-breakpoint-tablet {
        padding-right: $spacing-base;
        padding-left: $spacing-base;
      }
    }

    &--prose {
      max-width: $page-max-width-prose;
      padding-right: $spacing-lg;
      padding-left: $spacing-lg;

      @include bpk-breakpoint-tablet {
        padding-right: $spacing-base;
        padding-left: $spacing-base;
      }
    }

    &--center {
      display: flex;
      width: 100%;
      flex-direction: column;
      align-items: center;
      text-align: center;
    }
  }
}
