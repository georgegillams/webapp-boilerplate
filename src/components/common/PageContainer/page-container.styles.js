import PageContainer from '@george-gillams/components/page-container';
import styled, { css } from 'styled-components';

export const StyledPageContainer = styled(PageContainer)`
  ${({ debug }) =>
    debug &&
    css`
      background: linear-gradient(0.375turn, #ffdea3, orange);
    `}
`;
