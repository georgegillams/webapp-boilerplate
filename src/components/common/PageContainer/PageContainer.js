import React, { useState, useEffect } from 'react';

import { DEBUG_SHOW_PAGE_CONTAINER_KEY } from '@george-gillams/webapp/helpers/storageConstants';
export * from '@george-gillams/components/page-container';
import { StyledPageContainer } from './page-container.styles';

const PageContainer = props => {
  const [debug, setDebug] = useState(false);

  useEffect(() => {
    setDebug(window.localStorage.getItem(DEBUG_SHOW_PAGE_CONTAINER_KEY) === 'true');
  }, []);

  return <StyledPageContainer debug={debug} {...props} />;
};

export default PageContainer;
