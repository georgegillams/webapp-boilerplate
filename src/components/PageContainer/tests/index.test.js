import React from 'react';

import { render } from '@testing-library/react';

import PageContainer from '..';

describe('<PageContainer />', () => {
  const children = <div>some text</div>;

  it('Should render correctly', () => {
    const { container } = render(<PageContainer>{children}</PageContainer>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with custom className', () => {
    const { container } = render(<PageContainer className={'test-custom-classname'}>{children}</PageContainer>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with prose', () => {
    const { container } = render(<PageContainer prose>{children}</PageContainer>);

    expect(container).toMatchSnapshot();
  });
});
