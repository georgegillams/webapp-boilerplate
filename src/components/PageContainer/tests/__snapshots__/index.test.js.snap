// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContainer /> Should render correctly 1`] = `
<div>
  <main
    class="page-container__container"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;

exports[`<PageContainer /> Should render correctly with custom className 1`] = `
<div>
  <main
    class="page-container__container test-custom-classname"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;

exports[`<PageContainer /> Should render correctly with prose 1`] = `
<div>
  <main
    class="page-container__container page-container__container--prose"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;
